### PR TITLE
Social | Update upgrade nudges on settings pages

### PR DESCRIPTION
**Pull request by manzoorwanijk**
To account for the changes in Social plans (pdrWKz-1Du-p2), we need to update the upgrade nudges we show on the Social and Jetpack settings pages.

This PR preserves the backward compatibility for Advanced and Basic plans.

## Proposed changes:

* Create a `hasPaidFeatures` flag in the initial state
* Create a `hasPaidFeatures` selector in the store to keep the logic centralized instead of directly checking for the features. It allows us to change the value from the backend.
* Update the upgrade nudges to be displayed only when there are no paid features.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pdrWKz-1Du-p2

## Testing instructions:

- With Advanced Plan
    - Goto Jetpack settings > Sharing section
    - Confirm that you don't see the upgrade nudge
    - Goto Social admin page
    - Confirm that you don't see the upgrade nudge
- With Basic Plan
    - Goto Jetpack settings > Sharing section
    - Confirm that you DO see the upgrade nudge for Advanced plan
    - Goto Social admin page
    - Confirm that you don't see the upgrade nudge
- With the new Social plan
    - Goto Jetpack settings > Sharing section
    - Confirm that you don't see the upgrade nudge
    - Goto Social admin page
    - Confirm that you don't see the upgrade nudge

Screenshot 2024-05-30 at 1 27 55 PM
Screenshot 2024-05-30 at 1 28 30 PM

### CODE REVIEW

File: .vscode/settings.json
Comment (gmjuhasz) [inline .vscode/settings.json:1]: Is this intended?

Comment (manzoorwanijk) [reply .vscode/settings.json:1]: Yes, those are just lint fixes and it's annoying to always discard this file whenever I switch branches, stash/unstash changes. So it's better to fix those lints.

Comment (manzoorwanijk) [reply .vscode/settings.json:1]: Should I move that to its own PR?

Comment (manzoorwanijk) [reply .vscode/settings.json:1]: I am moving it to its own PR. thanks

**Comment by spsiddarthan** [comment]
This PR looks good on the first read, I haven't tested it yet though. If Geri doesn't get to it today, me and Geri can review and test this and ensure it gets merged by tomorrow.

**Review by gmjuhasz — APPROVED** [review]
This tests well

**Pull request merged**